src/Control/Distributed/Process/Node/RegistryAgent.hs

```diff
@@ -0,0 +1,81 @@
+{-# LANGUAGE BangPatterns, CPP #-}
+-----------------------------------------------------------------------------
+---- |
+---- Module      :  Control.Distributed.Process.Node.RegistryAgent
+---- Copyright   :  (c) Tweag I/O 2015
+---- License     :  BSD3 (see the file LICENSE)
+----
+---- Maintainer  :  Tim Watson <watson.timothy@gmail.com>
+---- Stability   :  experimental
+---- Portability :  non-portable (requires concurrency)
+----
+---- This module provides registry monitoring agent, implemented as
+---- a /distributed-process Management Agent/. Once 'node' starts it run this
+---- agent. Agent will monitor every remote process that was added to the
+---- local registry. So node will receive Died notification and remove process
+---- from registry in case if target process is dead.
+----
+-------------------------------------------------------------------------------
+
+module Control.Distributed.Process.Node.RegistryAgent
+    ( registryMonitorAgent
+    ) where
+
+import Control.Distributed.Process.Management
+import Control.Distributed.Process.Internal.Types
+import Control.Distributed.Process.Internal.Primitives
+import Data.Foldable (forM_)
+#if MIN_VERSION_containers(5,0,0)
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+#else
+import Data.Map (Map)
+import qualified Data.Map as Map
+#endif
```

Comment:
Scrictness is given up in the older versions. Isn't it better to use the lazy version always and strictify the code below if needed at some place? I cannot spot any place where it is needed though.